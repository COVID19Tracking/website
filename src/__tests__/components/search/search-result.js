import React from 'react'
import renderer from 'react-test-renderer'
import SearchResult from '~components/search/search-result'

describe('Components : Search: Results', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<SearchResult title="Test title" url="https://example.com" />)
      .toJSON()
    expect(tree).toMatchSnapshot()

    const treeWithChildren = renderer
      .create(
        <SearchResult title="Test title" url="https://example.com">
          <p>A child</p>
        </SearchResult>,
      )
      .toJSON()
    expect(treeWithChildren).toMatchSnapshot()

    const treeWithAuthorAndDate = renderer
      .create(
        <SearchResult
          title="Test title"
          url="https://example.com"
          author="An author"
          publishDate="April 23, 2020"
        />,
      )
      .toJSON()
    expect(treeWithAuthorAndDate).toMatchSnapshot()
  })
})
