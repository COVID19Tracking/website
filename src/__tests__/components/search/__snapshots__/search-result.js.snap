// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components : Search: Results renders correctly 1`] = `
<div
  className="result"
>
  <h4>
    <a
      href="https://example.com"
    >
      Test title
    </a>
  </h4>
  <div />
</div>
`;

exports[`Components : Search: Results renders correctly 2`] = `
<div
  className="result"
>
  <h4>
    <a
      href="https://example.com"
    >
      Test title
    </a>
  </h4>
  <div>
    <p>
      A child
    </p>
  </div>
</div>
`;

exports[`Components : Search: Results renders correctly 3`] = `
<div
  className="result"
>
  <h4>
    <a
      href="https://example.com"
    >
      Test title
    </a>
  </h4>
  <div />
  <div
    className="detail-text detailText"
  >
    <div
      className="resultDetails"
    >
      <span
        className="publishDate"
      >
        Updated 
        April 23, 2020
      </span>
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": " &middot; ",
          }
        }
      />
      An author
    </div>
  </div>
</div>
`;
